pages/business/[id].tsx

```diff
@@ -1,108 +1,83 @@
+import { ArrowLeft, PaperPlane } from '@sumup/icons';
 import Link from 'next/link';
 import { CardIconSvg } from 'src/pages/business/svg-icons';
-import styled from 'utils/styled';
 import { ExternalLinks } from 'src/components/ExternalLinks';
 import { Clap, Merchant } from 'utils/types';
 import getAbsoluteURL from 'utils/getAbsoluteURL';
 import { Theme } from '@sumup/design-tokens';
-import { groupBy } from 'lodash';
-
-const Wrapper = styled.div(
-  () => css`
-    max-width: 1170px;
-    margin: 0 auto;
-    display: flex;
-    position: relative;
-    padding-top: 46px;
-  `,
-);
-
-const StyledCard = styled(Card)(
-  () => css`
-    box-shadow: 0px 2px 8px rgba(0, 0, 0, 0.1);
-    border-radius: 8px;
-    max-width: 100%;
-    width: 375px;
-    overflow: hidden;
-    margin-right: 32px;
-  `,
-);
-
-const StyledSubheading = styled(SubHeading)(
-  () => css`
-    text-transform: none;
-    font-size: 20px;
-    line-height: 24px;
-    color: #666666;
-    font-weight: normal;
-    margin-bottom: 24px;
-  `,
-);
-
-const StyledDt = styled.dt`
-  font-weight: normal;
-  font-size: 16px;
-  line-height: 24px;
-  color: #666666;
-`;
-
-const StyledDd = styled.dd`
-  font-weight: normal;
-  font-size: 16px;
-  line-height: 24px;
-  color: #333333;
-  margin-bottom: 16px;
-`;
-
-const BackButton = styled(Button)`
-  border-radius: 0;
-  border: none;
-  top: -10px;
-  position: absolute;
-`;
-
-const StyledImage = styled(Image)`
-  border-radius: 8px;
-  width: 248px;
-  height: 148px;
-  object-fit: cover;
-`;
-
-const ClapButton = styled(Button)(
-  ({ theme, hasClapped }: { theme: Theme; hasClapped?: boolean }) => css`
-    padding: ${theme.spacings.byte};
-    ${hasClapped &&
-    css`
-      border-width: ${theme.borderWidth.mega};
-      border-color: ${theme.colors.warning};
-      opacity: 1 !important;
-    `}
-  `,
-);
+import { groupBy, orderBy } from 'lodash';
+import { PurpleButton } from 'src/components/PurpleButton';
+
+import { CommentRow } from '../../src/pages/business/comment';
+import {
+  Wrapper,
+  StyledCard,
+  BackButton,
+  ClapButton,
+  StyledDd,
+  StyledDt,
+  StyledImage,
+  StyledSubheading,
+} from '../../src/pages/business/business-page.styles';
+import { firebase } from '../../src/lib/firebase';
+import 'firebase/database';
 
 const BusinessPage = ({ merchant, id }: { merchant: Merchant; id: string }) => {
   const [claps, setClaps] = useState<{ [key: string]: Clap[] }>({});
-  const [isFiring, setFiring] = useState<boolean>(false);
+  const [isFiring, setFiring] = useState(false);
+  const [comment, setComment] = useState('');
+  const [comments, setComments] = useState(
+    orderBy(Object.values(merchant.comments || {}), 'dateAdded', 'desc'),
+  );
+
   const AuthUser = useAuthUser();
 
+  const handleFormSubmit = async (e: React.SyntheticEvent) => {
+    e.preventDefault();
+    if (!comment.trim().length) {
+      return;
+    }
+
+    const commentId = new Date().getTime();
+
+    await firebase
```

Comment:
Ah so we're doing the firebase DB calls inline again 😅 #hackweek

Reply:
yes :D :D 
I even think this is how it should work. It seems like it's safe to use the client only in front-end apps and `firebase-admin` for servers 😄

Reply:
yeah it's totally fine, I just like the idea of handling firebase init and API calls only from API routes, but 🤷 doesn't matter that much